From 7da974f9d4d28a4c87eff8326c7b78825bd461e9 Mon Sep 17 00:00:00 2001
From: Mark Gates <mgates3@utk.edu>
Date: Mon, 15 Jul 2024 22:41:05 +0000
Subject: [PATCH 02/10] cmake: handle CUDA arch >= 10.0, sm_xyz

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61c5e0b79..70da6c267 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,17 +173,16 @@ if (MAGMA_ENABLE_CUDA)
     endif()
 
     # MIN_SM is lowest requested version
-    set( MIN_SM 99 )
+    set( MIN_SM 9999 )
 
-    string( REGEX MATCHALL "sm_[0-9][0-9]" sms "${GPU_TARGET}" )
-    list( SORT sms ) # To make output easier to reason about
+    string( REGEX MATCHALL "sm_[0-9]+[0-9]" sms "${GPU_TARGET}" )
     foreach (sm IN LISTS sms)
         string( REPLACE "sm_" "" sm "${sm}") # Remove sm_ prefix
         if (sm LESS MIN_SM)
             set( MIN_SM "${sm}" )
         endif()
         list( APPEND __cuda_architectures "${sm}" )
-        string( REGEX REPLACE "([0-9])([0-9])" "\\1.\\2" cuda_arch "${sm}" )
+        string( REGEX REPLACE "([0-9]+)([0-9])" "\\1.\\2" cuda_arch "${sm}" )
         message( STATUS "    compile for CUDA arch ${cuda_arch}" )
     endforeach()
 
@@ -194,6 +193,7 @@ if (MAGMA_ENABLE_CUDA)
                  "Was: ${GPU_TARGET}" )
     endif()
 
+    message( STATUS "CUDA_ARCHITECTURES ${__cuda_architectures}" )
     set(CUDA_ARCHITECTURES "${__cuda_architectures}")
     set( MIN_ARCH ${MIN_SM}0 )
 
-- 
2.47.1

