From ae83cdd0c38e6de3d9b6ba1a33ae95e925a36f69 Mon Sep 17 00:00:00 2001
From: Ahmad Abdelfattah <ahmad@icl.utk.edu>
Date: Thu, 31 Oct 2024 10:50:47 -0400
Subject: [PATCH 10/10] add some missing cuda archs

---
 Makefile | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index d7273a5f0..fd5a174e4 100644
--- a/Makefile
+++ b/Makefile
@@ -139,21 +139,32 @@ ifeq ($(BACKEND),cuda)
 	endif
 	ifneq ($(findstring Maxwell, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_50
+		CUDA_ARCH_ += sm_52
+		CUDA_ARCH_ += sm_53
 	endif
 	ifneq ($(findstring Pascal, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_60
+		CUDA_ARCH_ += sm_61
+		CUDA_ARCH_ += sm_62
 	endif
 	ifneq ($(findstring Volta, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_70
+		CUDA_ARCH_ += sm_72
 	endif
 	ifneq ($(findstring Turing, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_75
 	endif
 	ifneq ($(findstring Ampere, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_80
+		CUDA_ARCH_ += sm_86
+		CUDA_ARCH_ += sm_87
+	endif
+	ifneq ($(findstring Ada, $(GPU_TARGET)),)
+		CUDA_ARCH_ += sm_89
 	endif
 	ifneq ($(findstring Hopper, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_90
+		CUDA_ARCH_ += sm_90a
 	endif
 
 
@@ -169,7 +180,7 @@ ifeq ($(BACKEND),cuda)
 	# See also $(info compile for ...) in Makefile
 
 
-    CUDA_ARCH_UNKNOWN_ = $(filter-out sm_% Kepler Maxwell Pascal Volta Turing Ampere Hopper, $(CUDA_ARCH_))
+    CUDA_ARCH_UNKNOWN_ = $(filter-out sm_% Kepler Maxwell Pascal Volta Turing Ampere Ada Hopper, $(CUDA_ARCH_))
     ifneq ($(CUDA_ARCH_UNKNOWN_),)
         $(error ERROR: unknown `$(CUDA_ARCH_UNKNOWN_)` in GPU_TARGET)
     endif
@@ -188,7 +199,7 @@ ifeq ($(BACKEND),cuda)
 
     # Check for empty
     ifeq ($(NV_SM),)
-        $(error ERROR: unknown `GPU_TARGET=$(GPU_TARGET)`. Set cuda_arch to one or more of Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Hopper, or valid sm_XX from nvcc -h)
+        $(error ERROR: unknown `GPU_TARGET=$(GPU_TARGET)`. Set cuda_arch to one or more of Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Ada, Hopper, or valid sm_XX from nvcc -h)
     else
         # Get last option (last 2 words) of nv_compute.
         nwords := $(words $(NV_COMP))
@@ -205,7 +216,7 @@ ifeq ($(BACKEND),cuda)
     CUDA_ARCH := $(SMS)
     CUDA_ARCH_MIN := $(word 1, $(SMS))0
     ifeq ($(CUDA_ARCH_MIN),)
-        $(error GPU_TARGET, currently $(GPU_TARGET), must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Hopper, or valid sm_[0-9][0-9]. Please edit your make.inc file)
+        $(error GPU_TARGET, currently $(GPU_TARGET), must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Ada, Hopper, or valid sm_[0-9][0-9]. Please edit your make.inc file)
     endif
 else ifeq ($(BACKEND),hip)
 
-- 
2.47.1

